### PR TITLE
Ct 2105 edit late responder team

### PR DESCRIPTION
**Pull request by starswan**
Allow editing of the team responsible for a case lateness when the closing data is being edited.

### CODE REVIEW

File: spec/features/cases/foi/editing_case_closure_spec.rb [hunk 1]
```diff
@@ -13,6 +13,16 @@
...
   end
 
+  scenario 'editing a late closed case', js: true do
+    kase = create :closed_case, :late
+
+    login_as manager
+    cases_show_page.load(id: kase.id)
+    edit_foi_case_closure_step(kase: kase,
+                               date_responded: 10.business_days.ago,
+                               late_team_id: (kase.transitions.map(&:acting_team_id).uniq - [kase.late_team_id]).first)
```
Comment (mseedat-moj) [inline spec/features/cases/foi/editing_case_closure_spec.rb]: Would be clearer to extract out `kase.transitions.map(&:acting_team_id).uniq - [kase.late_team_id]` to a variable IMO.

**Review by mseedat-moj — CHANGES_REQUESTED** [review]
Recommend test code has line extracted to variable for easier comprehension

File: spec/support/features/steps/cases/edit_steps.rb [hunk 2]
```diff
@@ -60,6 +63,13 @@ def edit_foi_case_closure_step(kase:, # rubocop:disable Metrics/MethodLength, Me
     .to eq kase.date_responded.month.to_s
   expect(cases_edit_closure_page.date_responded_year.value)
     .to eq kase.date_responded.year.to_s
+  if kase.responded_late?
+    expect(cases_edit_closure_page.get_late_team(kase.late_team_id)).to be_checked
+    expect(kase.late_team_id).not_to eq(late_team_id)
+    if late_team_id
```
Comment (mseedat-moj) [inline spec/support/features/steps/cases/edit_steps.rb]: When would `late_team_id` be false and why?

**Review by mseedat-moj — COMMENTED** [review]
Looks good, have a question regarding a test (see comment)

File: spec/support/features/steps/cases/edit_steps.rb (hunk 2 above)
Comment (starswan) [reply spec/support/features/steps/cases/edit_steps.rb]: it's an optional parameter

**Review by mseedat-moj — APPROVED** [review]

**Pull request merged**